test/test_sklearn/test_sklearn_train.py

```diff
@@ -0,0 +1,175 @@
+"""
+Tracking:
+* When failinga to import with a low memory limit, an import error is raised
+"""
+from __future__ import annotations
+
+from typing import Any
+
+import faulthandler
+import os
+import platform
+import time
+
+import numpy as np
+import psutil
+
+from pynisher import (
+    CpuTimeoutException,
+    MemoryLimitException,
+    WallTimeoutException,
+    contexts,
+    limit,
+    supports,
+)
+from pynisher.util import Monitor
+
+import pytest
+
+
+def train_svr(model: Any, X: np.ndarray, y: np.ndarray, limit: Any) -> bool:
+    """Trains an svr model"""
+    m = Monitor()
+    print("Memory before fit", m.memory("mb"))
+    print("Limit ", limit)
+
+    # This will cause a coredump with sigdev
+    with open(os.devnull, "rb") as f:
+        faulthandler.enable(file=f)
+        model.fit(X, y)
+    print("Memory after fit", m.memory("mb"))
```

Comment:
Is this line and the return still necessary given the coredump?

Reply:
Also, how are all the tests working if this gives a coredump?

Reply:
The return isn't necessary and the catching of the core dump isn't either, but it keeps the tests clean. Also in case the test fails, it's good debug information.

If a subprocess core dumps, it won't kill the parent process so it can happen safely in the tests.

Reply:
Maybe I'm missing something, but if there's a core dump the process is treated as failure, right? So this function is used in almost all scikit-learn tests, wouldn't this mean that all of these tests fail?

Reply:
It only happens when memory is limited so it only effects:
https://github.com/automl/pynisher/blob/f78f79a02237439b180c1f1a4cff185439988401/test/test_sklearn/test_sklearn_train.py#L49-L59